src/Parser/Serialization/FuncSerializer.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Text;
+
+namespace Parser
+{
+    public class FuncSerializer<T> : ISerialisation
+        where T : class //todo add also class for struct
+    {
+        private Func<T, string> _func;
+
+        public FuncSerializer(Func<T, string> func)
+        {
+            _func = func;
+        }
+
+        public void SerializeObject(StringBuilder sb, object value, IFormatProvider formatProvider)
+        {
+            var o = value as T;
+            if (o != null)
+            {
+                sb.Append(_func(o));
```

Comment:
The fact that the `formatProvider` is not used (I know it comes from the interface) kind of looks like an unexpected behavior to me.
Suggested implementation for this method:
```csharp
if (o is T)
  sb.Append(Convert.ToString(o, formatProvider));
```